src/core/coap/coap_base.hpp

```diff
@@ -152,6 +152,51 @@ class CoapBase
      */
     uint16_t GetPort(void) { return mSocket.GetSockName().mPort; };
 
+    /**
+     * This method sends a CoAP empty message.
+     *
+     * @param[in]  aType           The message type
+     * @param[in]  aRequestHeader  A reference to the CoAP Header that was used in CoAP request.
+     * @param[in]  aMessageInfo    The message info corresponding to the CoAP request.
+     *
+     * @retval kThreadError_None         Successfully enqueued the CoAP response message.
+     * @retval kThreadError_NoBufs       Insufficient buffers available to send the CoAP response.
+     * @retval kThreadError_InvalidArgs  The @p aRequestHeader header is not of confirmable type.
+     *
+     */
+    ThreadError SendEmptyMessage(Header::Type aType, const Header &aRequestHeader,
```

Comment:
Perhaps this function could be private, as we have specialized wrappers (SendReset/SendAck).